### PR TITLE
parse full duration for time formatting

### LINKED ISSUE #1495: Question: how to make PipelineRun show actual duration when it took longer than 1 hour?
In PipelineRun  

```bash
tekton$ tkn pr describe npu-tools-run-rsbj8
Name:              npu-tools-run-rsbj8
Namespace:         tekton
Pipeline Ref:      npu-tools
Service Account:   build-bot
Timeout:           4h0m0s
Labels:
 tekton.dev/pipeline=npu-tools

🌡️  Status

STARTED       DURATION   STATUS
2 hours ago   1 hour     Succeeded

📦 Resources

 NAME      RESOURCE REF
 ∙ image   image

⚓ Params

 NAME            VALUE
 ∙ gitRevision   next-gen-ci

📝 Results

 No results

📂 Workspaces

 NAME       SUB PATH   WORKSPACE BINDING
 ∙ source   ---        PersistentVolumeClaim (claimName=tekton2)

🗂  Taskruns

 NAME                                         TASK NAME              STARTED       DURATION     STATUS
 ∙ npu-tools-run-rsbj8-128dpes-integration    128dpes-integration    1 hour ago    1 hour       Succeeded
```

Task running time was longer than 1 hour but the description seems not to have actual value but hard-coded 1 hour. I could confirm that task ran over than 1 hour in dashboard, so wondered why it did not appear in the PipelineRun.

Is there any configuration to show actual value for it?

Thanks

### PR DESCRIPTION
**Pull request by jbpratt**
# Changes

Use native `.String()` impl for `time.Duration`

fixes #1495

Signed-off-by: jbpratt <jbpratt78@gmail.com>

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Run the code checkers with `make check`
- [x] Regenerate the manpages, docs and go formatting with `make generated`
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/cli/blob/master/CONTRIBUTING.md)
for more details._

# Release Notes

```release-note
Print more accurate duration
```

### CODE REVIEW

**Comment by jbpratt** [comment]
Hi, this may not be the desired result for describing the duration, if so we can close this and close up https://github.com/tektoncd/cli/issues/1495 as won't fix :smiley_cat:

File: pkg/formatted/time_test.go [hunk 1]
```diff
@@ -23,13 +23,76 @@ import (
...
+	}
+}
+
+func TestDuration(t *testing.T) {
+	now := time.Now()
+	startTime := metav1.NewTime(now)
+	plusFifteen := metav1.NewTime(now.Add(time.Minute * 15))
+	plusOneHour := metav1.NewTime(now.Add(time.Hour * 1))
+	plusOneHourAndHalf := metav1.NewTime(now.Add(time.Hour*1 + time.Minute*30))
+	plusOneDayAndHalf := metav1.NewTime(now.Add(time.Hour*24 + time.Minute*35))
```
Comment (jbpratt) [inline pkg/formatted/time_test.go:60]: There may be a cleaner way to do this, open for suggestions :)

**Comment by jbpratt** [comment]
I think the output in question (in the issue) comes from here https://github.com/tektoncd/cli/blob/d1bbad0e986cb3433b1cbba829e8959746963020/pkg/pipelinerun/description/description.go#L113-L119 which uses this function https://github.com/tektoncd/cli/blob/d1bbad0e986cb3433b1cbba829e8959746963020/pkg/pipelinerun/description/description.go#L192 leading to this https://github.com/tektoncd/cli/blob/d1bbad0e986cb3433b1cbba829e8959746963020/pkg/formatted/time.go#L32-L39 Changing `ParseShort` to `Parse` results in the seemingly desired output.

**Comment by jbpratt** [comment]
Adding some output examples:
```
❯ tkn taskrun list
NAME                                                            STARTED      DURATION     STATUS
zzzzz-6xz4m                                                     1 hour ago   1 hour       Succeeded

❯ ./bin/tkn taskrun list
NAME                                                            STARTED      DURATION                      STATUS
zzzzz-6xz4m                                                     1 hour ago   1 hour 30 minutes 6 seconds   Succeeded

❯ tkn taskrun describe zzzzz-6xz4m
...
STARTED      DURATION    STATUS
1 hour ago   1 hour      Succeeded
...

❯ ./bin/tkn taskrun describe zzzzz-6xz4m
...
STARTED      DURATION                      STATUS
1 hour ago   1 hour 30 minutes 6 seconds   Succeeded
```

which I think is closer to what is shown in the dashboard

**Comment by piyush-garg** [comment]
Instead of having this long text `1 hour 30 minutes 6 seconds`, I think we can have something like `1h30m6s`

**Comment by jbpratt** [comment]
@piyush-garg PTAL, I've updated to use the native `.String` impl for `time.Duration` but days are represented differently.

**Comment by jbpratt** [comment]
Again, if this isn't desired we can close this out and leave an explanation on the ticket (https://github.com/tektoncd/cli/issues/1495)

**Comment by jbpratt** [comment]
Hi @pradeepitm12 @piyush-garg @vdemeester please take a look

File: pkg/formatted/time_test.go (hunk 1 above)
Comment (chmouel) [reply pkg/formatted/time_test.go:60]: there is not many ways to skin that cat :D

File: pkg/formatted/time.go [hunk 2]
```diff
@@ -35,7 +35,7 @@ func Duration(t1, t2 *metav1.Time) string {
 	}
 
 	dur := t2.Time.Sub(t1.Time)
-	return durafmt.ParseShort(dur).String()
```
[the remark this answers is not in the document]
Comment (jbpratt) [reply pkg/formatted/time.go:38]: So `durafmt` is still used below in the `Timeout` function https://github.com/tektoncd/cli/blob/a79baaef8cea48a2bdc93d98048a47f3158854d6/pkg/formatted/time.go#L46 but we can probably update this as well. I'll take a look after work and see if this makes sense.

[the remark this answers is not in the document]
Comment (chmouel) [reply pkg/formatted/time.go:38]: cool, but i am fine having this in a follow up,

thanks for checking,

**Pull request merged**